app/javascript/src/Stats/inlinechart/index.jsx

```diff
@@ -0,0 +1,137 @@
+// @flow
+
+type Props = {
+    endPoint: string,
+    metricName: string,
+    title: string,
+    unitPluralized: string
+}
+
+type State = {
+    loading: boolean,
+    title: string,
+    total: string,
+    values: Array<number>,
+    unit: string,
+    unitPluralized: string
+}
+
+type RefObject = { current: null | HTMLDivElement }
+
+type DataMetric = {
+    unit: string
+}
+
+type Data = {
+    metric: DataMetric,
+    total: number,
+    values: Array<number>
+}
+
+type ResponseError = {
+  status: string,
+  statusText: string
+}
+
+import React, {Component} from 'react'
+import c3 from 'c3'
+import 'whatwg-fetch'
+import 'core-js/es6/promise'
+import 'url-polyfill'
+import moment from 'moment'
+
+class InlineChart extends Component<Props, State> {
+  c3ChartContainer: RefObject
+
+  constructor (props: Props) {
+    super(props)
+    this.c3ChartContainer = React.createRef()
+    this.state = {
+      loading: true,
+      title: this.props.title,
+      total: '',
+      values: [],
+      unit: '',
+      unitPluralized: this.props.unitPluralized
+    }
+  }
+
+  generateC3Chart () {
+    const nodeElem = this.c3ChartContainer.current
+    c3.generate({
+      bindto: nodeElem,
+      axis: {
+        x: { show: false },
+        y: { show: false }
+      },
+      legend: { show: false },
+      point: { show: false },
+      data: {
+        columns: [
+          [...this.state.values]
+        ]
+      },
+      tooltip: {
+        contents: function (d) {
+          return `<span><i class='tooltip-dot'></i> ${d[0].value}</span>`
+        }
+      },
+      onresize: function () {
+        nodeElem.style.maxHeight = 'none'
+      }
+    })
+  }
+
+  getTotalAsString (total: number, unit: string) {
+    return `${total.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')} ${unit.substring(0, 10)}`
+  }
+
+  updateState (data: Data) {
+    let unit = data.metric.unit
+    if (unit.slice(-1) !== 's' && data.total !== 1) {
+      unit = this.state.unitPluralized
+    }
+    const total = this.getTotalAsString(data.total, unit)
+
+    this.setState({
+      loading: false,
+      total,
+      unit,
+      values: data.values
+    }, () => this.generateC3Chart())
+  }
+
+  getURL () {
+    const { endPoint, metricName } = this.props
+    const today = moment(new Date())
+    const until = today.format('YYYY-MM-DD')
+    const since = today.subtract(30, 'day').format('YYYY-MM-DD')
+    const granularity = 'day'
+    return new URL(`${endPoint}?metric_name=${metricName}&since=${since}&until=${until}&granularity=${granularity}`, window.location.origin)
```

Comment:
You could use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#Example, it's safer and then a better assertion could be made for testing